src/seedu/addressbook/common/Utils.java

```diff
@@ -34,4 +38,15 @@ public static boolean elementsAreUnique(Collection<?> items) {
         }
         return true;
     }
+
+    /**
+     * Return true if two set of strings are not disjoint, regardless of capitalization.
+     */
+    public static boolean isCaseInsensitiveMatched(Set<String> firstSet, Set<String> secondSet) {
```

Comment:
Good job on extracting the method. Maybe u can extract it even more by having a method that compares the strings themselves (instead of just the sets), and use that within this method.